From 2d71b4a8e7ffad4bca4729ceee674089903cc24a Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 8 Aug 2017 13:06:06 +0800
Subject: [PATCH 2347/5242] MLK-16158-2 ARM64: dts: imx8mq-evk: add a new dts
 to support 'LCDIF-->DSI'

commit  ebbc2d64a1c5a08d466a053ea5eb31b77ccddc84 from
https://source.codeaurora.org/external/imx/linux-imx.git

There are two independent display path on imx8mq platform:

    a. DCSS  --> HDMI
    b. LCDIF --> DSI

The default enabled display path is 'DCSS-->HDMI', so add
a new dts to enable 'LCDIF-->DSI', and at the same time,
disable 'DCSS-->HDMI' path.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8mq-evk-lcdif-dsi.dts     |   31 ++++++++++++++++++++
 2 files changed, 33 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-dsi.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index d58d477..8f4ada8 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -30,4 +30,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-wm8962.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb \
 				 fsl-imx8mq-evk-od.dtb \
-				 fsl-imx8mq-evk-4k.dtb
+				 fsl-imx8mq-evk-4k.dtb \
+				 fsl-imx8mq-evk-lcdif-dsi.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-dsi.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-dsi.dts
new file mode 100644
index 0000000..0d89242
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-dsi.dts
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+&lcdif {
+	status = "okay";
+};
+
+&mipi_dsi {
+	status = "okay";
+};
+
+&dcss {
+	status = "disabled";
+};
+
+&hdmi {
+	status = "disabled";
+};
-- 
1.7.9.5

